sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/MultiSpanProcessor.java

```diff
@@ -68,11 +69,17 @@ public boolean isEndRequired() {
 
   @Override
   public CompletableResultCode shutdown() {
-    List<CompletableResultCode> results = new ArrayList<>(spanProcessorsAll.size());
-    for (SpanProcessor spanProcessor : spanProcessorsAll) {
-      results.add(spanProcessor.shutdown());
+    synchronized (this) {
```

Comment:
`synchronized (this)` is always a bad idea, as you open yourself for deadlock if somebody external decides to synchronise on this object instance. In this particular case I think `AtomicBoolean` is better than `volatile+lock`

Reply:
updated